api4/channel.go

```diff
@@ -115,6 +117,26 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPublicChannels(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Minor but can we rename this to `getPublicChannelsForTeam` so it's a bit more explicit? (Same with the client4.go and store functions)